sunkit_image/reproject.py

```diff
@@ -0,0 +1,106 @@
+"""
+Functionality for reprojecting solar images between different coordinate
+systems.
+"""
+import numpy as np
+from reproject import reproject_adaptive, reproject_exact, reproject_interp
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+from astropy.wcs import WCS
+from sunpy.map import Map, make_fitswcs_header
+from sunpy.sun import constants
+
+__all__ = ["reproject_carrington"]
+
+_reproject_funcs = {"interp": reproject_interp, "exact": reproject_exact, "adaptive": reproject_adaptive}
+
+
+def reproject_carrington(smap, shape_out, latitude_projection="CAR", *, reproject_method="interp", **reproject_kwargs):
```

Comment:
Given the recent addition of `GenericMap.reproject_to` should we be encouraging the use of `smap.reproject_to(carrington_heasder(...))` instead of this function?

or put a different way: what's this making easier than that?

Reply:
I think reprojecting to a Carrington frame (as opposed to a generic frame) is a common enough task that it deserves to have a really easy way to do it, ie.

```python
car_map = reproject_carrington(smap, shape_out)
```
vs.
```python
header = carrington_header(smap.date, shape_out, smap.observer_coordinate)
car_map = smap.reproject_to(WCS(header))
```

Reply:
I see your point, but why not have this API:

```python
wcs = carrington_wcs_from_map(smap, shape_out)
car_map = smap.reproject_to(wcs)
```

I feel like keeping the call to `reproject_to` maintains awareness that this is a thing you can do to any map and wcs and it simplifies the API by separating the two parts, i.e. you don't need to pass through all the reproject kwargs and the args for making the output wcs at the same time etc?

Reply:
Also as an aside: why can't we pass a header into `reproject_to`?

Reply:
I'm 👍 to that, didn't think about improving `carrington_header_from_map`. And maybe we can pass a header to `reproject_to`? I forget...

Reply:
Of course you can supply a header.  What do you take me for, a monster? =P